test/apps/test_fuzzy_match.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/python2
+
+import os
+import commands
+import re
+import argparse
+import sys
+import logging
+
+###################
+# Main test runner
+###################
+class FuzzyTester:
+    def __init__(self, ucx_info, verbose):
+        self.ucx_info = ucx_info
+        if verbose:
+            logging.basicConfig(level=logging.DEBUG)
+        
+    def cleanup(self):
+        ucx_vars = [var for var in os.environ.keys() if var.startswith('UCX_')]
+        for var in ucx_vars:
+            del os.environ[var]
+
+    def run(self, env_vars):
+        self.cleanup()
+        logging.info('Using env vars: %s' % env_vars.keys())
+        
+        matches = self.get_fuzzy_matches(env_vars)
+
+        if matches != env_vars:
+            raise Exception('wrong fuzzy list: got: %s, expected: %s' % (matches, env_vars))
+        
+        logging.info('found all expected matches: %s' % env_vars)
+        
+    def exec_ucx_info(self):
+        cmd = self.ucx_info + ' -u m -w'
+        logging.info('running cmd: %s' % cmd)
+    
+        status, output = commands.getstatusoutput(cmd)
+        if status != 0:
+            raise Exception('Received unexpected exit code from ucx_info: ' + str(status))
+            
+        logging.info(output)
+        return output
+    
+    def get_fuzzy_matches(self, env_vars):
+        for var_name in env_vars:
+            os.environ[var_name] = 'value'
+        
+        output = self.exec_ucx_info()
+        warn_msg = output.splitlines()[0]
+        
+        warn_match = re.match('.*Detected unused environment variables?: (.*)', warn_msg)
+        if not warn_match:
+            raise Exception('"unused vars" message was not found')
+            
+        output_vars = warn_match.group(1).split(';')
+        matches = [re.match(r'(\w+)(?: \(did you mean: (.*)\?\))?', var.strip()) for var in output_vars]
+        if None in matches:
+            raise Exception('Unexpected warning message format: %s' % warn_msg)
+        
+        return {m.group(1) : [x.strip() for x in m.group(2).split(',')] if m.group(2) else [] for m in matches}
+    
+    
+def parse_ucx_info(path):
```

Comment:
Does this func "parse ucx_info"? Maybe rename.

Reply:
WDYTA getting the full path to `ucx_info` as an input parameter and therefore just get rid of this function?

Reply:
we'll need this function, because you need to verify the path actually exists.
But I agree it's better to pass the path to the binary.

Reply:
Ok, but IMO in this case it rather `check_ucx_info_path` than `pare_ucx_info`